lib/access.go

```diff
@@ -0,0 +1,93 @@
+package lib
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/qri-io/qri/auth/token"
+	"github.com/qri-io/qri/profile"
+)
+
+// AccessMethods is a group of methods for access control & user authentication
+type AccessMethods struct {
+	d dispatcher
+}
+
+// Name returns the name of this method group
+func (m AccessMethods) Name() string {
+	return "access"
+}
+
+// Access returns the authentication that Instance has registered
+func (inst *Instance) Access() AccessMethods {
+	return AccessMethods{d: inst}
+}
+
+// CreateAuthTokenParams are input parameters for Access().CreateAuthToken
+type CreateAuthTokenParams struct {
+	GranteeUsername  string
+	GranteeProfileID string
+	TTL              time.Duration
+}
+
+// SetNonZeroDefaults uses default token time-to-live if one isn't set
+func (p *CreateAuthTokenParams) SetNonZeroDefaults() {
+	if p.TTL == 0 {
+		p.TTL = token.DefaultTokenTTL
+	}
+}
+
+// Valid checks if the profile in question is valid
+func (p *CreateAuthTokenParams) Valid() error {
```

Comment:
Small nitpick: `Valid` reads to me like a yes/no question, ie something that would return a `bool`. Could we rename this method to `Validate` instead? There's a bunch of other places in the codebase that use this name for methods that return an error on invalid structs, which feels like a good pattern to keep.

Reply:
ah totally agreed. Will change